Content.Server/Store/Systems/StoreSystem.cs

```diff
@@ -87,6 +88,10 @@ private void OnAfterInteract(EntityUid uid, CurrencyComponent component, AfterIn
         if (!TryComp<StoreComponent>(args.Target, out var store))
             return;
 
+        // PDAs without uplinks shouldn't be allowed to have currency added
+        if (TryComp<RingerComponent>(args.Target, out var ringer) && !TryComp<RingerUplinkComponent>(args.Target, out var ringerUplink))
```

Comment:
Like I said in the technical details header, I don't really like how this is done. While it _works_ this features PDA uplink code in the generic store code, which is a bit anti-OOP. However, I'm limited by the fact that I don't know the codebase well enough to do it in a better way. Suggestions or pointers for an improved way are appreciated.